e2e-tests/stream-mode-tests.sh

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env bash
+
+#######################################
+# Function that prints messages
+# Arguments:
+#   anything that needs printing.
+#######################################
+function print_message() {
+  local print_prefix="E2E: STREAM MODE TEST:"
+  echo "${print_prefix} $*"
+}
+
+#######################################
+# Call common-mode-tests to setup 
+# the testing env
+#  
+#######################################
+source ./e2e-tests/common-mode-tests.sh
+setup
+
+print_message "OPENMRS AND FHIR SERVER STARTED SUCCESSFULLY"
+
+test_dir=$(mktemp -d -t analytics_tests__XXXXXX)
+
+print_message "PARQUET FILES WILL BE WRITTEN INTO $test_dir DIRECTORY"
+
+print_message "BUILDING THE ANALYTICS  PIPELINE"
+
+timeout 1m mvn compile exec:java -pl streaming-binlog \
+   -Dexec.args="--openmrsUserName=admin --openmrsPassword=Admin123 \
+   --openmrsServerUrl=http://localhost:8099/openmrs \
+   --openmrsfhirBaseEndpoint=/ws/fhir2/R4 \
+   --outputParquetPath='${test_dir}' --fhirSinkPath=http://localhost:8098/fhir --sinkUserName=hapi --sinkPassword=hai "
+
+
+print_message "SEARCH INDEX REBUILD"
+
+curl -i -u admin:Admin123 -X POST http://localhost:8099/openmrs/ws/rest/v1/searchindexupdate
```

Comment:
yes ,we need to do the searchindex rebuild because when when openmrs instance is up, it shows no records inside , so we do the search index manually in order to be able to get the resources needed to POST  the data

Reply:
Sorry, just to make sure I understand: Are you saying that building the search index is needed for the POST to work? Or are you saying it is only needed to be able to see the patients when searching for them in OpenMRS, e.g., through the UI?

BTW, do you know how long this takes?

Reply:
yes the search index is needed for the POST to work ,,,, so that i can get the data needed to do the posting  ,,,going to check out the time it takes and report back

Reply:
I'm with Bashir on being confused about this. Which POST request fails without this? (So I'm clear, my understanding is that we're really only indexing a very small number of things in Lucene; mostly concepts and patients; those indexes shouldn't be necessary to read data via the FHIR module and I'm not sure why they'd be necessary for creating an encounter or visit...)

Reply:
initially the postinf of a patient was failing but now its working even without the search index,,,,
thanks folks for the correction